tests/sentry/receivers/test_default_detector.py

```diff
@@ -0,0 +1,214 @@
+
+    def test_creates_detector_without_team(self):
+        """Test that detector can be created without an owner team."""
+        project = self.create_project()
+
+        with mock.patch("sentry.workflow_engine.processors.detector.send_new_detector_data"):
+            detector = _ensure_metric_detector(project, owner_team_id=None, enabled=True)
+
+        assert detector.owner_team_id is None
+        assert detector.enabled is True
+
+    def test_send_new_detector_data_failure_does_not_block_creation(self):
+        """Test that detector is still created even if sending data to Seer fails."""
+        project = self.create_project()
+
+        with mock.patch(
+            "sentry.workflow_engine.processors.detector.send_new_detector_data",
+            side_effect=Exception("Seer unavailable"),
+        ):
+            detector = _ensure_metric_detector(project)
+
+        assert detector is not None
+        assert Detector.objects.filter(id=detector.id).exists()
+
+    def test_returns_existing_detector_without_creating_duplicates(self):
+        """Test that calling _ensure_metric_detector twice returns the same detector."""
+        project = self.create_project()
+
+        with mock.patch("sentry.workflow_engine.processors.detector.send_new_detector_data"):
+            detector1 = _ensure_metric_detector(project)
+            detector2 = _ensure_metric_detector(project)
+
+        assert detector1.id == detector2.id
+        assert Detector.objects.filter(project=project, type=MetricIssue.slug).count() == 1
+        assert DataSource.objects.filter(organization_id=project.organization_id).count() == 1
+        assert QuerySubscription.objects.filter(project=project).count() == 1
+
+
+class TestCreateMetricDetectorWithOwner(TestCase):
+    @with_feature(
+        {
+            "organizations:default-anomaly-detector": True,
+            "organizations:anomaly-detection-alerts": True,
+        }
+    )
+    def test_creates_enabled_detector_when_both_features_enabled(self):
+        """Test that detector is created and enabled when both feature flags are enabled."""
+        from sentry.receivers.project_detectors import create_metric_detector_with_owner
+
+        project = self.create_project()
+        # Get the team that was auto-created with the project
+        team = project.teams.first()
+        assert team is not None
+
+        with mock.patch("sentry.workflow_engine.processors.detector.send_new_detector_data"):
+            create_metric_detector_with_owner(project, user=self.user)
+
+        detector = Detector.objects.get(project=project, type=MetricIssue.slug)
+        assert detector.name == "High Error Count (Default)"
+        assert detector.owner_team_id == team.id
+        assert detector.enabled is True
+
+    @with_feature(
+        {
+            "organizations:default-anomaly-detector": True,
+            "organizations:anomaly-detection-alerts": False,
+        }
+    )
```

Comment:
```suggestion
    @with_feature("organizations:default-anomaly-detector")
```

`anomaly-detection-alerts` is already False by default